staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -281,6 +281,21 @@ func AddAuditAnnotation(ctx context.Context, key, value string) {
 	addAuditAnnotationLocked(ac, key, value)
 }
 
+// AddAuditAnnotationForRejectWithReason records an audit annotation
+// indicating that a watch request was rejected for the given reason.
+func AddAuditAnnotationForRejectWithReason(ctx context.Context, reason string) {
+	AddAuditAnnotation(ctx, "audit.k8s.io/watch-reject-reason", reason)
+}
+
+// AddAuditAnnotationForRejectMessage adds a human-readable message annotation
+// truncated to 1024 characters to avoid excessive log size.
+func AddAuditAnnotationForRejectMessage(ctx context.Context, msg string) {
+	if len(msg) > 1024 {
+		msg = msg[:1024] + "…"
+	}
+	AddAuditAnnotation(ctx, "audit.k8s.io/watch-reject-message", msg)
```

Comment:
Let's pick something in `openshift.io` for the namespace part of the key instead.